iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -214,61 +214,222 @@ inline bool string<Capacity>::unsafe_assign(const std::string& str) noexcept
 }
 
 template <uint64_t Capacity>
-template <uint64_t N>
-inline int64_t string<Capacity>::compare(const string<N>& other) const noexcept
+template <typename T>
+inline
+    typename std::enable_if<(is_cxx_string<T>::value || is_char_array<T>::value || std::is_same<T, std::string>::value),
+                            int64_t>::type
+    string<Capacity>::compare(const T& other) const noexcept
 {
-    uint64_t otherSize = other.size();
-    if (m_rawstringSize < otherSize)
+    uint64_t otherSize = internal::GetSize<T>::call(other);
+    auto result = memcmp(c_str(), internal::GetData<T>::call(other), std::min(m_rawstringSize, otherSize));
```

Comment:
One could argue it does not matter here if we do not have early `\0` characters in our string. I am not sure we allow those. If we do, `memcmp` is required as `strncmp` stops at the first `\0` character (or if the n characters where compared).

So they behave differently but it is only relevant if we can have a string with `\0` characters, otherwise they should behave the same and theoretically `memcmp` is better than as it will not check for `\0`.

TLDR: if my assumption regarding `\0` chars is correct, `memcpy` should be preferred for performance.